src/client/app/components/meters/MeterModalEditComponent.tsx

```diff
@@ -0,0 +1,567 @@
+/* eslint-disable @typescript-eslint/indent */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react';
+import { Modal, Button, Dropdown } from 'react-bootstrap';
+import { GPSPoint } from 'utils/calibration';
+import '../../styles/meter-edit-modal.css';
+
+interface meterTypes {
+  MAMAC?: string;
+  METASYS?: string;
+  OBVIUS?: string;
+  OTHER?: string
+}
+interface MetaDataProps {
+  onhide: () => void;
+  onSaveChanges: () => void;
+  handleIdentifierChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  onCancel: () => void;
+  show: boolean;
+  id: number;
+  identifier: string;
+  name: string;
+  units?: string;
+  meterType?: string;
+  gps?: GPSPoint;
+  Area: number;
+  displayable: boolean;
+  enabled: boolean;
+  meterAddress?: string;
+  graphicUnit: number;
+  notes: string;
+  cumulative: boolean;
+  cumulativeReset: boolean;
+  cumulativeResetStart: string;
+  cumulativeResetEnd: string;
+  endOnlyTime: boolean;
+  readingGap: string;
+  readingVariation: number;
+  readingDuplication: number;
+  timesort: boolean;
+  startTimestamp: string;
+  endTimestamp: string;
+}
+
+type MeterViewPropsWithIntl = MetaDataProps;
+
+class MeterModalEditComponent extends React.Component<MeterViewPropsWithIntl,
+  { show: boolean, nameinput: string, identifierinput: string, identifier: string }>{
+  constructor(props: MeterViewPropsWithIntl) {
+    super(props);
+    this.state = {
+      show: this.props.show,
+      nameinput: this.props.name,
+      identifier: this.props.identifier,
+      identifierinput: this.props.identifier
+    };
+  }
+
+  render() {
+    return (
+      <>
```

Comment:
It isn't needed, when we were first making the MeterModalEditComponent.tsx we were using it as a "boiler" of sorts so that if anything broke within the code within that return type, the modalPage would still function. We removed it and tested the functionality. No errors have occured and has the same functions with or without it.